File: vol1/ch1/LeetCode1287.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        n=len(arr)
        c,cnt=arr[0],0
        for  i in range(0,n):
            if arr[i]==c:
                cnt+=1
                if cnt*4>n: return c
            else:
                c,cnt=arr[i],1

File: vol1/ch1/LeetCode1287.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        int n=arr.size();
        int c=arr[0], cnt=0;
        for (int i=0;i<n;i++) {
            if (arr[i]==c) {
                cnt++;
                if (cnt*4>n) return c;
            }
            else {
                c=arr[i];
                cnt=1;
            }
        }
        return -1;
    }
};

